hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/dml/TestMergeIntoTable.scala

```diff
@@ -263,6 +340,137 @@ class TestMergeIntoTable extends HoodieSparkSqlTestBase with ScalaAssertionSuppo
     })
   }
 
+  test("Test MergeInto with changing partition and global index") {
+    withRecordType()(withTempDir { tmp =>
+      withSQLConf("hoodie.index.type" -> "GLOBAL_SIMPLE") {
+        Seq("cow","mor").foreach { tableType => {
+          val sourceTable = generateTableName
+          val targetTable = generateTableName
+          spark.sql(
+            s"""
+               | create table $sourceTable
+               | using parquet
+               | partitioned by (partition)
+               | location '${tmp.getCanonicalPath}/$sourceTable'
+               | as
+               | select
+               | 1 as id,
+               | 2 as version,
+               | 'yes' as mergeCond,
+               | '2023-10-02' as partition
+            """.stripMargin
+          )
+          spark.sql(s"insert into $sourceTable values(2, 2, 'no', '2023-10-02')")
```

Comment:
can we make tests simpler to read. 
for eg, why we insert into diff partitions. 
can we insert to same partition for all 3 record keys. 
and later ingest 1 of the record to new partition. 2nd one to same partition and leave the 3rd w/o touching. and also insert a new record key. 

the cur test is bit hard to follow